### DISCUSSION TITLE
Add cheat that disables ride crashes from train collisions

### DISCUSSION THREAD

**Discussion by MarcelVos96** [OpenRCT2/OpenRCT2#21273] [Feature requests]
Some ride vehicles, such as the log flume, can never crash, no matter how fast the boats collide. A cheat that would do the same for rollercoasters and other rides would allow for all kinds of crazy shenanigans.

Crashes from derailing trains don't count for this. You could disable the ones that happen when you take a turn or hill too fast, but I don't see how you could ever disable crashes for when the train just runs off the end of the track.

**Comment by pfroud** [comment]
I think the starting point is this flag getting set in these three places:

`UpdateTrackMotionForwards()`:
https://github.com/OpenRCT2/OpenRCT2/blob/704333344e9254b38fbdc4eaef93c07cc56ec9c5/src/openrct2/ride/Vehicle.cpp#L7842


`UpdateTrackMotionBackwards()`:
https://github.com/OpenRCT2/OpenRCT2/blob/704333344e9254b38fbdc4eaef93c07cc56ec9c5/src/openrct2/ride/Vehicle.cpp#L8150

and...`UpdateTrackMotionMiniGolfVehicle()`??
https://github.com/OpenRCT2/OpenRCT2/blob/704333344e9254b38fbdc4eaef93c07cc56ec9c5/src/openrct2/ride/Vehicle.cpp#L8643

**Comment by TheClaudeQC** [comment]
what if the was 4 different states in the cheat menu?

Disabled : no ride will crash
Flying-off only : only the rides that can fly off the track by being too fast will crash (if they fly-off)
Default : Like we have right now
Extreme : All rides can crash by collisions and the merry-go-round will crash after 5 minutes whiteout fixing a control failure
